Remotion/ObjectBinding/Web/UI/Controls/Validation/ValidationFailureHandlingContext.cs

```diff
@@ -0,0 +1,52 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// using NUnit.Framework;
+//
+using System.Text;
+using Remotion.ObjectBinding.Web.UI.Controls.BocListImplementation;
+using Remotion.Utilities;
+
+namespace Remotion.ObjectBinding.Web.UI.Controls.Validation
+{
+  public class ValidationFailureHandlingContext
+  {
+    public IBocList BocList { get; }
+
+    public IBocListValidationFailureRepository ValidationFailureRepository => BocList.ValidationFailureRepository;
+
+    private readonly StringBuilder _sb;
+
+    public ValidationFailureHandlingContext (IBocList bocList)
+    {
+      ArgumentUtility.CheckNotNull("bocList", bocList);
+
+      BocList = bocList;
+      _sb = new StringBuilder();
+    }
+
+    public string GetCompoundedErrorMessage () => _sb.ToString();
```

Comment:
After reading the rest of the PR, in particular the usage pattern of this method:
I think storing the messages in a List<string>, passing the StringBuilder into an `AppendErrorMessages(StringBuilder)` method would be better. That way we do not need to copy the strings twice and also have less memory allocations.